### PR TITLE
Add "-i" to sendmail's pipe

### LINKED ISSUE #16379: Email Logo - two dots in url
### Steps to reproduce
1. Send Test Mail in E-Mail Server Settings
2. Picture link in mail is broken - checking URL Shows:
https://subdomain..com/index.php/apps/theming/image/logo?useSvg=0&v=7
>> There are two dots behind the Domain Name.
>> Opening the link with one dot > logo was shown up

Testing on 2 fresh Setups / new database with 16.0.3 > same issue as in updated nextcloud
Cannot find any rewrite issue with a wrong Domain.

### LINKED ISSUE #33754: First email notification link wrong
Dear NextCloud team,

We've got a NextCloud instance into a subdomain mysub.mydomain.com. I've checked the config.php and everything seems good.
However, the first email for a new member refers to a link with mysub..mydomain.com (double dots).

Is there a way to fix it?

Regards,
Fabien

### LINKED ISSUE #39594: [Bug]: Double dots in links in emails
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

As you can from this issue in Symfony https://github.com/symfony/symfony/issues/50875 seems like the mailer adds double dots if a dot is matched as first character of a line (this is the code)  
https://github.com/symfony/symfony/blob/acd9e3b9922b2f6064dc8bd1f77c52f7ce48ec69/src/Symfony/Component/Mailer/Transport/SendmailTransport.php#L105-L107

If you add the `-i` in the `pipe` (like this `'pipe' => ' -t -i'`)
https://github.com/nextcloud/server/blob/767c42f4596fc077480b3729b344c9a261cabe9c/lib/private/Mail/Mailer.php#L333-L336
This fixes the issue.  

Symphony community is still trying to understand if this is a bug or not, but still this seems a solution.  
What do you think? Can I do a PR for this?

This will resolve these issues too: #33754 and #16379

### Steps to reproduce

1. replace the sendmail binary with the postfix one (I don't know if this is required but I use this one) but stop all postfix daemons (or just run it in a container with a super dummy spool folder)
```bash
mkdir -p /var/spool/postfix/maildrop
chmod ugo+rwx  /var/spool/postfix /var/spool/postfix/maildrop
```
3. set the `mail_sendmailmode` to `pipe`
4. create many users with different string length (for example user1, user12, user123, user1234)
```bash
# create random users
export OC_PASS=Epi1cP4ssword
for i in {1..20}; do php occ user:add user$(python3 -c "print('1'*$i)") --password-from-env; done
# set the email
for i in {1..20}; do USERNAME=user$(python3 -c "print('1'*$i)"); php occ user:setting "$USERNAME" settings email "${USERNAME}.${USERNAME}@example.com"; done
```
5. generate some emails (for example I send all the Welcome emails with the reset password link)
6. check if there are duplicated dots in the postfix spool (grep random things that could appear in your Nextcloud url)
```bash
cd /var/spool/postfix/maildrop
postcat -bh * | grep lostpass -C2
```
7. you will definitely find at least one user with a broken link 
```
Imposta la tua password: https://mydrive.example.com/index=
..php/lostpassword/reset/form/3GOnnGyVaEBzs2JmuYXCU/utente123456
```

### Expected behavior

Dots should not be doubled in emails

### Installation method

Community Docker image

### Nextcloud Server version

26

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
[...]
        "mail_smtpmode": "sendmail",
        "mail_sendmailmode": "pipe",
[...]
    }
}
```

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by FedericoHeichou**
* Resolves: #39594 
* Fix https://github.com/nextcloud/server/issues/33754
* Fix https://github.com/nextcloud/server/issues/16379

## Summary
Symfony Mailer replaces `"\n."` with `"\n.."`.  
If the email to be sent contains links when they are breaklined a line may start with a `.`, thus can generating a URL like `https://www..example.com/[...]` or `https://www.example.com/index..php/[...]` .  
With "-i" this replacement is not performed.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by FedericoHeichou** [comment]
Hello, do I need to do something?

**Comment by tanganellilore** [comment]
Hi team,
I'm also interested to this features... @solracsf @icewind1991 can you help us to address this and include in 28?

Thanks

**Comment by joshtrichards** [comment]
**TL;DR** I agree that `-i` should probably be added, but in theory it's not supposed to matter given [this bit of code](https://github.com/symfony/symfony/blob/62e115f5db9836ea5821b97fabbd2ef74491bdfd/src/Symfony/Component/Mailer/Transport/SendmailTransport.php#L105-L106). But, even so, it should be added because doing is is probably the Right Thing(tm) to do for the broadest compatibility. It's also how I read [Symfony's inline recommendation](https://github.com/symfony/symfony/blob/62e115f5db9836ea5821b97fabbd2ef74491bdfd/src/Symfony/Component/Mailer/Transport/SendmailTransport.php#L43-L50).

I'm wondering if what's happening is a one (or both) of:
- some third-party command-line option "sendmail compatibility layers" are accepting `-i` as a parameter but not actually handling mail in "`-i` mode" (thus creating a problem with `.`
- some third-party command-line option "sendmail compatibility layers" are operating in `-i` mode by default even when the option hasn't been specified (thus triggering Symfony's code above unintentionally and undesirably because Symfony merely looks for the presence or lack there of of `-i` on the command-line to decide whether to adjust message content).

My money is on the second being more common.

The balance - probably the bulk since this would be a much more widely reported otherwise I'd imagine - are handling things just fine *without* `-i` being present by operating how an official sendmail binary would when not provided with that option (and thus the activation of the Symfony code above is both desirable and functioning).

Bottom line: It's completely reasonable to run `-t` mode with or without the `-i` flag when using the Symfony Mailer. (Well, as long as that bit of code linked above is working).

IMO: Since our aim should be compatibility (and as deterministic as possible outcomes) we should:

- explicitly specify `-i` when operating in `-t` mode (what Symfony calls "pipe" mode)

Ideally, [documentation](https://github.com/nextcloud/documentation) also gets a minor addition in the Server Requirements or Email Configuration section: e.g. *If using pipe mode we'll be invoking the installed sendmail binary (genuine or compatible). We expect the called sendmail binary to fully support the following conventional sendmail compatible options: `-bs`, `-i`, `-t`, and `-f`. These are generally supported universally since they're central to sendmail compatibility, but occasionally there have been differences in behavior. These differences can cause problems ranging from inability to send mail to the mangling of periods in message bodies. If you run into issues such as inability to send email or unusual email mangling, please try to troubleshoot your sendmail installation or try switching to `smtp` mode.*

All that said: There may yet be further reports of issues like this (but hopefully fewer). My recollection from my email server days is that the only dot `.` that matters with sendmail/smtp is the one that is the very first character of a line. A cursory look at #33754, #39594, and #16379 doesn't make it immediately apparent to me why these particular message lines were identified for adjustment by [this code](https://github.com/symfony/symfony/blob/62e115f5db9836ea5821b97fabbd2ef74491bdfd/src/Symfony/Component/Mailer/Transport/SendmailTransport.php#L105-L106) regardless of the presence (or not) of `-i`. Of course it's also possible they were mangled downstream - by the server's own mailer for whatever reason. :man_facepalming: 

Still same conclusion: we should send `-i`.

I offered you the *TL;DR*. You didn't have to read this far. :innocent:

**Review by joshtrichards — APPROVED** [review]
LGTM

**Review by kesselb — APPROVED** [review]

**Comment by joshtrichards** [comment]
No idea why the cypress tests keep failing for this PR. 

```
 1) Versions expiration
       Expire all versions:
     CypressError: `cy.exec('docker exec --user www-data nextcloud-cypress-tests-server php ./occ config:system:set versions_retention_obligation --value "0, 0"')` failed because the command exited with a non-zero code.

Pass `{failOnNonZeroExit: false}` to ignore exit code failures.

Information about the failure:
Code: 1

Stdout:
An unhandled exception has been thrown:
Error: Call to undefined method OC\Server::getEventDispatcher() in /var/www/html/console.php:94
Stack trace:
#0 /var/www/html/occ(11): require_once()
#1 {main}

https://on.cypress.io/exec
      at <unknown> (http://172.17.0.2/__cypress/runner/cypress_runner.js:119190:77)
      at tryCatcher (http://172.17.0.2/__cypress/runner/cypress_runner.js:1807:23)
      at Promise._settlePromiseFromHandler (http://172.17.0.2/__cypress/runner/cypress_runner.js:1519:31)
      at Promise._settlePromise (http://172.17.0.2/__cypress/runner/cypress_runner.js:1576:18)
      at Promise._settlePromise0 (http://172.17.0.2/__cypress/runner/cypress_runner.js:1621:10)
      at Promise._settlePromises (http://172.17.0.2/__cypress/runner/cypress_runner.js:1701:18)
      at _drainQueueStep (http://172.17.0.2/__cypress/runner/cypress_runner.js:2407:12)
      at _drainQueue (http://172.17.0.2/__cypress/runner/cypress_runner.js:2400:9)
      at Async._drainQueues (http://172.17.0.2/__cypress/runner/cypress_runner.js:2416:5)
      at Async.drainQueues (http://172.17.0.2/__cypress/runner/cypress_runner.js:2286:14)
  From Your Spec Code:
      at Context.eval (webpack://nextcloud/./cypress/support/commands.ts:160:7)

  2) Versions expiration
       Expire versions v2:
     CypressError: `cy.exec('docker exec --user www-data nextcloud-cypress-tests-server php ./occ config:system:set versions_retention_obligation --value "0, 0"')` failed because the command exited with a non-zero code.

Pass `{failOnNonZeroExit: false}` to ignore exit code failures.

Information about the failure:
Code: 1

Stdout:
An unhandled exception has been thrown:
Error: Call to undefined method OC\Server::getEventDispatcher() in /var/www/html/console.php:94
Stack trace:
#0 /var/www/html/occ(11): require_once()
#1 {main}
```
